pyiceberg/table/update/__init__.py

```diff
@@ -727,6 +728,11 @@ class AssertRefSnapshotId(ValidatableTableRequirement):
     ref: str = Field(...)
     snapshot_id: Optional[int] = Field(default=None, alias="snapshot-id")
 
+    @model_serializer(mode="wrap")
+    def serialize_model(self, handler) -> dict[str, Any]:
+        partial_result = handler(self)
+        return {**partial_result, "snapshot-id": self.snapshot_id}
```

Comment:
nit: lets add back the comment to show why this is needed
```suggestion
    @model_serializer(mode="wrap")
    def serialize_model(self, handler) -> dict[str, Any]:
        # Ensure "snapshot-id" is always present, even if value is None
        partial_result = handler(self)
        return {**partial_result, "snapshot-id": self.snapshot_id}
```